modules/fedimint-wallet-client/src/lib.rs

```diff
@@ -398,10 +406,19 @@ impl WalletClientModule {
                             .get_deposit_address_inner(valid_until, &mut dbtx.module_dbtx())
                             .await;
 
-                        // Begin watching the script address
-                        self.rpc
-                            .watch_script_history(&address.script_pubkey())
-                            .await?;
+                        let script = &address.script_pubkey();
+                        if !already_watched
+                            .lock()
+                            .expect("locking failed")
+                            .contains(script)
+                        {
+                            // Begin watching the script address
+                            self.rpc.watch_script_history(script).await?;
+                            already_watched
+                                .lock()
+                                .expect("locking failed")
+                                .insert(script.clone());
+                        }
```

Comment:
How likely is this retry to happen? Should only occur if the `get_deposit_address` fn is called highly concurrently. So is the additional code complexity worth it?

Reply:
I think I was wrong and it is very unlikely. Will drop it.